src/main/java/com/github/davidcarboni/thetrain/api/Commit.java

```diff
@@ -27,75 +32,117 @@
     @POST
     public Result commit(HttpServletRequest request,
                          HttpServletResponse response) throws IOException, FileUploadException {
-
-        Log.info("Start Commit");
         Transaction transaction = null;
+        String transactionId = null;
         String message = null;
-        boolean error = false;
+        boolean isError = false;
 
         try {
 
             // Transaction ID
-            String transactionId = request.getParameter("transactionId");
+            transactionId = request.getParameter(TRANSACTION_ID_KEY);
             if (StringUtils.isBlank(transactionId)) {
+                warn("commit: transactionID required but none provided")
+                        .log();
                 response.setStatus(HttpStatus.BAD_REQUEST_400);
-                error = true;
+                isError = true;
                 message = "Please provide a transactionId parameter.";
             }
 
             // Transaction object
-            String encryptionPassword = request.getParameter("encryptionPassword");
+            String encryptionPassword = request.getParameter(ENCRYPTION_PASSWORD_KEY);
+            if (StringUtils.isEmpty(encryptionPassword)) {
+                warn("commit: encryptionPassword required but none was provided")
+                        .transactionID(transactionId)
+                        .log();
+                response.setStatus(HttpStatus.BAD_REQUEST_400);
+                isError = true;
+                message = "encryptionPassword required but was empty or null " + transactionId;
+            }
```

Comment:
Seems odd that if the transaction ID or encryption key are not provided it doesn't return, it carries on without it. I am not suggesting you change this as I see it was like this before and you intentionally don't want to change functionality. May be worth keeping a note of and changing in future though.

Reply:
Yeah I thought the same. IMO it should be fail fast - if they are null or missing then that info is potentially lost or overwritten with another message giving false info. For now I am leaving it as is but given the priority call I would gladly change it.